torch/csrc/autograd/profiler.cpp

```diff
@@ -519,17 +554,61 @@ bool profilerEnabled() {
   return state_ptr && state_ptr->config().state != ProfilerState::Disabled;
 }
 
+bool kinetoAvailable() {
+#ifdef USE_KINETO
+  return true;
+#else
+  return false;
+#endif
+}
+
+void prepareProfiler(
+    const ProfilerConfig& new_config,
+    const std::set<ActivityType>& activities) {
+#ifdef USE_KINETO
+  if (new_config.state == ProfilerState::KINETO) {
+    std::set<libkineto::ActivityType> k_activities;
+    if (activities.count(ActivityType::CPU)) {
+      k_activities.insert(libkineto::ActivityType::EXTERNAL_CORRELATION);
```

Comment:
We probably want this together with CUDA_RUNTIME. I need to add another type for PyTorch events.

Reply:
Actually, this should be enabled if both CUDA_RUNTIME and CUDA is selected, as it provides links between those layers.

Reply:
This provides links between CUDA runtime events and PyTorch observer events, so once adding a new type for PyTorch observer events (e.g. APPLICATION or CLIENT) then this can be enabled when both that and CUDA_RUNTIME is selected.
Another use case where it may be useful is if you want to focus only on PyTorch and GPU events, removing the CUDA_RUNTIME layer, but keeping the links. We can then link directly to PyTorch events. We are still required to collect runtime events to do this unfortunately so it won't reduce overhead, only trace size.